api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -764,10 +771,59 @@ public User setUserActivationKey(User user) {
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
 			
-		//Send Email with unhashed  activation key and Date:
-		
+		//Send Email with unhashed  activation key 
+		sendMail(user, token);
 		return user;
 	}
+	
+	public void sendMail(User user, String token) {
```

Comment:
You don't need this method and all this code, I did mention there is a messaging service and you should use it to send the email.

Reply:
This should be a one line call

Reply:
Hello @wluyima  I've been looking at the MessageService to use  that service method and send the email but. when i tried it fails with errors of **Must issue a STARTTLS command first. v10-v6sm19344157wrm.18 - gsmtp** and i guess the MessageService is for only notification and alerts. as when i looked at the MailMessageSender. it has methods setMailSession which takes a session variable and creating a session requires the Propertites and an authenticator which is what i did in the code above. I was hopping to see some where. how they use the Global variables defined in the OpenmrsConstants. but i couldn't find any.
Thanks for any directions.

Reply:
I did use Context.getMessageService().sendMessage(mail) where mail is a message object ; and since no session is set message always fails

Reply:
There is a sendMessage method that doesn't take a message object, I would try that first.

Reply:
The failure is when am still building so by then am not yet logged in. i will do some digging again and see.